packages/core/src/Table/TableRow.js

```diff
@@ -1,45 +1,63 @@
 import propTypes from '@dhis2/prop-types'
 import cx from 'classnames'
-import React, { useContext } from 'react'
-import css from 'styled-jsx/css'
-import { TableContext } from './TableContext'
-
-const tableRowStyles = css`
-    .zebraStriping:nth-child(even) {
-        background: #fbfcfd;
-    }
-`
+import React, { useState, forwardRef } from 'react'
+import { DragHandleCell } from './TableRow/DragHandleCell.js'
+import { ExpandedRowContent } from './TableRow/ExpandedRowContent.js'
+import styles from './TableRow/TableRow.styles.js'
 
 /**
  * @module
  * @param {TableRow.PropTypes} props
  * @returns {React.Component}
- * @example import { TableRow } from '@dhis2/ui-core'
- * @see Live demo: {@link /demo/?path=/story/table--static-layout|Storybook}
+ * @example import { TableRow } from '@dhis2/ui'
+ * @see Live demo: {@link /demo/?path=/story/table--default|Storybook}
  */
-export const TableRow = ({
-    children,
-    className,
-    dataTest,
-    suppressZebraStriping,
-}) => {
-    const {
-        suppressZebraStriping: suppressZebraStripingFromContext,
-    } = useContext(TableContext)
-
-    const zebraStriping =
-        typeof suppressZebraStriping !== 'undefined'
-            ? !suppressZebraStriping
-            : !suppressZebraStripingFromContext
+export const TableRow = forwardRef(
+    (
+        {
+            children,
+            className,
+            dataTest,
+            expandableContent,
+            selected,
+            draggable,
+        },
+        ref
+    ) => {
+        const [
+            isHoveringExpandedContent,
+            setIsHoveringExpandedContent,
+        ] = useState(false)
```

Comment:
I think we could get rid of this if we implemented what I'm suggesting in the main comment

Reply:
We tried this approach and discussed on Slack. We discovered that for tables using an automatic table-layout, this approach causes issues with column width. Specifically, the columns of the nested table will completely ignore the column layout in the parent table. As such we concluded that the semantically imperfect solution is what we'd go with for now.